ONSdigital/eq-questionnaire-runner
# Simplify Progress Store structure

### What is the context of this PR?
Currently, the progress store contains duplication of `section_id` and `list_item_id` in the state. This makes the state larger than it needs to be. This PR removes these redundant data.
Instead of storing the full location object, only the block ids are now stored. I have also removed list names as we currently do not use them.

### How to review 
- Ensure survey progress works as expected.
- Ensure the new structure is:
```
{
      "section_id": "a-section",
      "block_ids": [
        "block-one",
        "block-two",
      ],
      "status": "IN_PROGRESS",
      "list_item_id": "dEUuoF"
}
```
